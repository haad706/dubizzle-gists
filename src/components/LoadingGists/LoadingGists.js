import PropTypes from "prop-types";
import styled from "styled-components";

export const LoadingGists = ({ count }) => {
  /**
   * skeletons can be increased/descreased using count
   */
  return (
    <Wrapper>
      {Array(count)
        .fill("")
        .map((_) => (
          <Skeleton>Loading...</Skeleton>
        ))}
    </Wrapper>
  );
};

const Skeleton = styled.div`
  box-shadow: 0px 2px 16px rgba(0, 0, 0, 0.15);
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  display: flex;
  border-radius: 10px;
  height: 170px;
  background-color: #efefef;
  align-items: center;
  justify-content: center;
`;

const Wrapper = styled.div`
  max-width: 700px;
  margin: 80px auto 0;
`;

LoadingGists.defaultProps = {
  count: 2,
};

LoadingGists.propTypes = {
  count: PropTypes.number,
};

export default LoadingGists;
