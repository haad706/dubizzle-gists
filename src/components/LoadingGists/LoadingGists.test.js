import { render } from "@testing-library/react";
import LoadingGists from "./LoadingGists";

describe("LoadingGists", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<LoadingGists />);
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<LoadingGists />);
    expect(asFragment()).toMatchSnapshot();
  });
});
