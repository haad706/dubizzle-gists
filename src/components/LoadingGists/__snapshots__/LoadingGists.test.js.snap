// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingGists should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gTRrQi jdwaWp"
  >
    <div
      class="sc-bgqQcB gXlruc"
    >
      Loading...
    </div>
    <div
      class="sc-bgqQcB gXlruc"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
